Simbody/include/simbody/internal/CableSpan.h

```diff
@@ -0,0 +1,446 @@
+        const;
+
+    /** Compute the frenet frame associated with the obstacle's curve segment at
+    the initial contact point on that obstacle.
+    If the path is not in contact with the obstacle's surface the frame will
+    contain invalid data. The frenet frame is measured relative to ground, with
+    the tangent along the X axis, the surface normal along the Y axis and the
+    binormal along the Z axis.
+    State must be realized to Stage::Position.
+    @param state State of the system.
+    @param ix The index of the obstacle in this CableSpan.
+    @return The frenet frame at the obstacle's initial contact point. **/
+    Transform calcCurveSegmentInitialFrenetFrame(
+        const State& state,
+        CableSpanObstacleIndex ix) const;
+
+    /** Compute the frenet frame associated with the obstacle's curve segment at
+    the final contact point on that obstacle.
+    If the path is not in contact with the obstacle's surface the frame will
+    contain invalid data. The frenet frame is measured relative to ground, with
+    the tangent along the X axis, the surface normal along the Y axis and the
+    binormal along the Z axis.
+    State must be realized to Stage::Position.
+    @param state State of the system.
+    @param ix The index of the obstacle in this CableSpan.
+    @return The frenet frame at the obstacle's final contact point. **/
+    Transform calcCurveSegmentFinalFrenetFrame(
+        const State& state,
+        CableSpanObstacleIndex ix) const;
+
+    /** Get the arc length of the obstacle's curve segment.
+    Returns NaN if the obstacle is not in contact with the path.
+    State must be realized to Stage::Position.
+    @param state State of the system.
+    @param ix The index of the obstacle in this CableSpan.
+    @return The arc length. **/
+    Real calcCurveSegmentArcLength(
+        const State& state,
+        CableSpanObstacleIndex ix) const;
+
+    ///@}
+
+    /** @cond **/ // Hide from Doxygen.
+    class Impl;
+
+private:
+    const Impl& getImpl() const
+    {
+        return *m_Impl;
+    }
+
+    Impl& updImpl()
+    {
+        return *m_Impl;
+    }
+
+    std::shared_ptr<Impl> m_Impl;
+
+    friend CableSubsystem;
+
+    // Befriend the helper class for testing the implementation.
+    friend CableSubsystemTestHelper;
+    /** @endcond **/
+};
+
+//==============================================================================
+/** This subsystem manages cables spanning between two points in a system.
+Each cable is represented by a CableSpan. Obstacles can be added to the cable,
+and must be wrapped over. The calculated path will consist of a series of
+straight line segments between obstacles, and curved segments (geodesics) over
+the obstacles. Force elements defined elsewhere may make use of the cables to
+apply forces to the system.
+
+See CableSpan. **/
+class SimTK_SIMBODY_EXPORT CableSubsystem : public Subsystem {
+public:
+    CableSubsystem();
```

Comment:
This constructor is also valid? Or should it be noted that the resulting system isn't connected to the multibody system (so, e.g. `getMultibodySystem` shall throw?)

Reply:
good point. Seems a bit useless: removed it.